src/main/java/org/jetbrains/bsp/bazel/server/bsp/resolvers/actiongraph/ActionGraphV1Parser.java

```diff
@@ -67,7 +49,14 @@ private String getTargetId(String needle) {
         .collect(Collectors.toList());
   }
 
-  private List<AnalysisProtos.Artifact> expandDepsetToArtifacts(Queue<String> idsToExpand) {
+  private List<AnalysisProtos.Artifact> expandDepsetToArtifacts(String idToExpand) {
+    Queue<String> idsToExpand =
```

Comment:
Can an interable be used if you're updating the queue whilst iterating through it?

Reply:
I meant just creating it using this method, I thought it could be used like [this](https://guava.dev/releases/19.0/api/docs/com/google/common/collect/Lists.html#newArrayList(E...)), not sure why there's no analogous method here, but still I'm not sure what this

```
{
            add(idToExpand);
          }
}
```

does exactly? Why can't it be a simple one-ine initialisation?

Reply:
Technically this is just a way of defining a series of operations that are related to the initialization. It's more semantics than anything. But I can use a list, sure.